File: programmers/js/level00/120875.js
// 평행

function solution(dots) {
  const caseArr = [
    [0, 1, 2, 3],
    [0, 2, 1, 3],
    [0, 3, 1, 2],
  ];

  for (const [idx1, idx2, idx3, idx4] of caseArr) {
    if (
      calcGradient(dots[idx1], dots[idx2]) ===
      calcGradient(dots[idx3], dots[idx4])
    ) {
      return 1;
    }
  }

  return 0;
}

function calcGradient(arr1, arr2) {
  return (arr2[1] - arr1[1]) / (arr2[0] - arr1[0]);
}

File: programmers/java/level00/120875.java
// 평행

class Solution {
    double calcGradient(int[] arr1, int[] arr2) {
        return (arr2[1] - arr1[1]) / (double) (arr2[0] - arr1[0]);
    }
    
    public int solution(int[][] dots) {
        int[][] caseArr = {
            {0, 1, 2, 3},
            {0, 2, 1, 3},
            {0, 3, 1, 2}
        };
        
        for (int[] idx : caseArr) {
            if (calcGradient(dots[idx[0]], dots[idx[1]]) == 
                calcGradient(dots[idx[2]], dots[idx[3]])) {
                return 1;
            }
        }
        
        return 0;
    }
}